### PR TITLE
adds git stash

### LINKED ISSUE #3708: [Entry] Subject: Stash under git
### Reason/inspiration (optional)

I would like to add information about the use of `git stash` in GitHub. 
We use `git stash` whenever we want to switch between the branches with the changes being done. Stash saves our work in a working directory. So, our changes doesn't get lost. 
Kindly assign this to me under hacktoberfest. 

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Vidip-Ghosh**
### Description

Closes #3708 

I've added the use of `git stash` command. This command is used when we have changes in the current branch and want to switch to different branch. So we stash the changes and switch to different branch. We run `git stash apply` to get the changes back. 

### Type of Change

- Adding a new entry
 Editing an existing entry (fixing a typo, bug, issues, etc)
 Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/git/concepts/stash/stash.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+Title: 'stash'
+Description: 'The changes made to the current working file are temporarily stored using the git stash command, which makes it simple to switch to a different branch and work on different features without losing modifications and apply the modifications again with git stash apply.'
```
Comment (avdhoottt) [inline content/git/concepts/stash/stash.md:3]:
```suggestion
Description: 'The changes made to the current working file are temporarily stored using the git stash command, which makes it simple to switch to a different branch and work on different features without losing modifications.
```

File: content/git/concepts/stash/stash.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---
+Title: 'stash'
```
Comment (avdhoottt) [inline content/git/concepts/stash/stash.md:2]:
```suggestion
Title: 'Stash'
```

File: content/git/concepts/stash/stash.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Bash/Shell'
+  - 'Developer tools'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Git'
+  - 'Version Control'
+  - 'GitHub'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-git'
+  - 'paths/computer-science'
```
Comment (avdhoottt) [inline content/git/concepts/stash/stash.md:13]:
```suggestion
Tags: 
  - 'Git'
  - 'Version Control'
  - 'GitHub'
CatalogContent: 
  - 'learn-git'
  - 'paths/computer-science'
```

File: content/git/concepts/stash/stash.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+Git **stash** temporarily stores the changes made in the current working file to the hidden directory called .git/refs/stash and we can use git **stash** apply to restore the changes done.
+
```
Comment (avdhoottt) [inline content/git/concepts/stash/stash.md:17]:
```suggestion
The **`git stash`** temporarily stores the changes made in the current working file to the hidden directory called .git/refs/stash and we can use **`git stash apply`** to restore the changes done.

```

File: content/git/concepts/stash/stash.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+new file:
+content/git/concepts/stash/stash.md
+Changes not staged for commit:
+(use
+"git add «file>...
+" to update what will be committed)
+(use
+"git restore «file>..."
+to
+```
```
Comment (avdhoottt) [inline content/git/concepts/stash/stash.md:47]: Add the complete output from the command line, this is incomplete.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Vidip-Ghosh Firstly, delete the yarn.lock file as there is no need to push it. I have also mentioned some changes, please make those.

**Comment by Vidip-Ghosh** [comment]
Why is it giving me conflicts? 
I try to integrate my changes but main branch shows something different.

**Comment by SSwiniarski** [comment]
@Vidip-Ghosh you accidently duplicated an earlier issue that someone had worked on and was merged. Apologies, your issue shouldn't have been accepted as it was a duplicate. We'll have to close this PR and the associated issue.

**Pull request closed without merging**